tasks/travis-pr-title-lint.js

```diff
@@ -0,0 +1,159 @@
+        if (err) {
+          reject(err);
+        } else {
+          resolve(stdout);
+        }
+      });
+    });
+  }
+
+  function getPullRequestTitle() {
+    return new Promise(function(resolve, reject) {
+      var pullRequestURLPath = `/repos/${ process.env.TRAVIS_REPO_SLUG
+        }${/pulls/ }${process.env.TRAVIS_PULL_REQUEST }.json`;
+
+      grunt.log.writeln(
+        `Retrieving the pull request title from https://api.github.com${
+          pullRequestURLPath}`
+      );
+
+      https.get({
+        host: 'api.github.com',
+        path: pullRequestURLPath,
+        headers: {
+          'User-Agent': 'mozilla web-ext grunt tasks',
+        },
+      }, function(response) {
+        var body = '';
+        response.on('data', function(data) {
+          body += data;
+        });
+        response.on('error', function(err) {
+          grunt.log.writeln(`Failed during pull request title download: ${
+                            err}`);
+          reject(err);
+        });
+        response.on('end', function() {
+          try {
+            resolve(JSON.parse(body).title);
+          } catch (err) {
+            reject(err);
+          }
+        });
+      });
+    });
+  }
+
+  function lintMessage(message) {
+    return new Promise(function(resolve, reject) {
+      return Promise.all([
+        getPreset('angular'),
+        getConfiguration('.conventional-changelog-lintrc'),
+      ]).then((results) => {
+        var preset = results[0];
+        var configuration = results[1];
+
+        grunt.log.writeln(`Changelog lint input: ${ message}`);
+        return changelogLint(message, {preset, configuration});
+      }).then((report) => {
+        const formatted = format(report, {
+          color: true,
+          signs: [' ', '⚠', '✖'],
+          colors: ['white', 'yellow', 'red'],
+        }).join('\n');
+
+        if (report.valid) {
+          resolve(formatted);
+        } else {
+          reject(formatted);
+        }
+      });
+    });
+  }
+
+  var writeCommitMessagesDocURL =
+        'https://github.com/mozilla/web-ext/blob/master/CONTRIBUTING.md#writing-commit-messages'; // eslint-disable-line max-len
+
+  grunt.registerTask(
+    'travis-pr-title-lint',
+    'get the pull request title in a travis build', function() {
+      if (process.env.TRAVIS_PULL_REQUEST &&
+          process.env.TRAVIS_PULL_REQUEST !== 'false') {
+        var done = this.async();
+
+        // Install ES6/ES7 polyfills required to use changelog-lint programmatically.
+        if (!Object.values) {
+          objectValues.shim();
+        }
+        if (!Object.entries) {
+          objectEntries.shim();
+        }
+
+        countGitMergeCommits().then(function(commitsCount) {
+          if (commitsCount === 1) {
+            grunt.log.writeln(
+              'There is only one commit in this pull request, ' +
+                'we are going to check the single commit message...'
```

Comment:
nit: typically we've been indenting breaks like this at the same level since it's not a nested block, it's a continuation of the block on the previous line